### PR TITLE
fix: [CDS-83240]: add type check before performing action on input

### PR DESCRIPTION
**Pull request by shivanand-harness**
### Summary

Jira Ticket: https://harness.atlassian.net/browse/CDS-83240

Actual issue: TypeError in /ng/account/bdW0lqdKTwWvq4YBjGfOVw/home/orgs/MOLOCO/projects/RMP_PoC/environments
e.map is not a function

Fix: added type check before performing action on input data

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentsFilters/filterUtils.ts [hunk 1]
```diff
@@ -74,6 +74,7 @@ export const getFilterByIdentifier = (identifier: string, filters?: FilterDTO[])
   filters?.find((filter: FilterDTO) => filter.identifier?.toLowerCase() === identifier.toLowerCase())
 
 export const getMultiSelectFromOptions = (values?: any[]): SelectOption[] | undefined => {
+  if (!Array.isArray(values)) return []
   return values?.map(item => {
```
Comment (sunnykesh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentsFilters/filterUtils.ts:78]: Will not just changing values?.map(item => {})  to values?.map?.(item => {}) will work?

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentsFilters/filterUtils.ts [hunk 2]
```diff
@@ -74,6 +74,7 @@ export const getFilterByIdentifier = (identifier: string, filters?: FilterDTO[])
   filters?.find((filter: FilterDTO) => filter.identifier?.toLowerCase() === identifier.toLowerCase())
 
 export const getMultiSelectFromOptions = (values?: any[]): SelectOption[] | undefined => {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentsFilters/filterUtils.ts:76]: We can add the proper type for values

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentsFilters/filterUtils.ts (hunk 1 above)
Comment (shivanand-harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentsFilters/filterUtils.ts:78]: yes this also works

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentsFilters/filterUtils.ts (hunk 2 above)
Comment (shivanand-harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentsFilters/filterUtils.ts:76]: Yes, I tried adding types, but this function is being called from multiple places with different types. if we add types here, then we need to convert everywhere before calling this function

**Review by pankajsingh-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Pull request merged**